parking/meter.py

```diff
@@ -2,48 +2,36 @@
 
 import csv
 
-PARKING_METER_LOCATION_DATA_FILE = "data/treas_parking_meters_loc_datasd.csv"
+from constants import GASLAMP_BOUNDING_BOX, PARKING_METER_LOCATION_DATA_FILE
 
-GASLAMP_BOUNDING_BOX = (32.716, -117.164, 32.705749, -117.157)
 
-		
 class Meter:
-
-	def __init__(self, dict):
-		self.id = dict['pole']	
-		self.long = float(dict['longitude'])
-		self.lat = float(dict['latitude'])
-
-
-	@classmethod
-	def allpoles(cls):
-		if not hasattr(cls, '_poles'):
-			poles = []
-
-			with open(PARKING_METER_LOCATION_DATA_FILE, 'r') as csvfile:
-				reader = csv.DictReader(csvfile)
-				poles = [Meter(row) for row in reader]
-						
-			cls._poles = poles
-			
-		return cls._poles
-
-
-	@classmethod
-	def pole_for_location(cls, pole):
-		if not hasattr(cls, '_poles_by_location'):
-			poles = {x['pole'] : x for x in cls.allpoles()}
-			cls._poles_by_location = poles
-
-		return cls._poles_by_location[pole] if pole in cls._poles_by_location else None
-		
-		
-	@classmethod
-	def poles_in_region(cls, latA, longA, latB, longB):
-		return list(filter(lambda l: l.lat <= latA and l.lat >= latB and
-								 	 l.long >= longA and l.long <= longB, cls.allpoles()))
-
-	@classmethod
-	def poles_in_gaslamp(cls):
-		return cls.poles_in_region(*GASLAMP_BOUNDING_BOX)
-
+    def __init__(self, data):
+        self.id = data['pole']
+        self.lng = float(data['longitude'])
+        self.lat = float(data['latitude'])
+
+    @classmethod
+    def allpoles(cls):
+        if not hasattr(cls, '_poles'):
+            poles = []
+
+            with open(PARKING_METER_LOCATION_DATA_FILE, 'r') as csvfile:
+                reader = csv.DictReader(csvfile)
+                poles = [Meter(row) for row in reader]
+
+            cls._poles = poles
+
+        return cls._poles
+
+    @classmethod
+    def poles_in_region(cls, bounding_box):
+        lat_a, lng_a, lat_b, lng_b = bounding_box
+        return [pole for pole in cls.allpoles() if (pole.lat <= lat_a and
```

Comment:
One of the common changes was translating filter/map/reduce. I think that these functions are confusing and not that easy to read. Even Guido has said that he dislikes them: https://www.artima.com/weblogs/viewpost.jsp?thread=98196

Reply:
I prefer filter/map/reduce as they have equivalent functions in many other languages so its easy to explain the concept and keywords, and apply them outside of just Python.

Reply:
I would argue the opposite. If someone is familiar with filter/map, they should be able to understand the list comprehension easily. And if they've never heard of filter/map, they're more readable :)

Reply:
The thing is that list comprehension doesn't exist as a fundamental building block in some other languages.  Check this out in JavaScript for example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Array_comprehensions  That said, I agree there are places where list comprehension is certainly more readable.  I wish there was a syntactic sugar for reducing the repetitiveness of `pole` in that expression.